src/sentry/static/sentry/app/views/organizationGroupDetails/reprocessingForm.tsx

```diff
@@ -3,115 +3,100 @@ import {browserHistory} from 'react-router';
-                {strong: <strong />}
-              )}
-            </ListItem>
-            <ListItem>
-              {' '}
-              {tct(
-                '[strong:Store Native crash reports to reprocess Minidump crash reports.] Note that this requires attachment storage.',
-                {strong: <strong />}
-              )}
-            </ListItem>
-            <ListItem>
-              {tct(
-                '[strong:Reprocessed events count towards your organization’s quota]. Rate limits and spike protection don’t apply here.',
-                {strong: <strong />}
-              )}
-            </ListItem>
-            <ListItem>
-              {tct('Wait one hour before attempting to reprocess missing debug files.', {
-                strong: <strong />,
-              })}
-            </ListItem>
-            <ListItem>
-              {tct(
-                'Reprocessed events will not trigger issue alerts, and reprocessed events are not subject to [link:data forwarding].',
-                {
-                  fwd: (
-                    <ExternalLink href="https://docs.sentry.io/platform-redirect/?next=/data-management/data-forwarding/" />
-                  ),
-                }
-              )}
-            </ListItem>
+          </Introduction>
+          <StyledList symbol="bullet">
+            {impacts.map((impact, index) => (
+              <ListItem key={index}>{impact}</ListItem>
+            ))}
           </StyledList>
           <ApiForm
-            apiEndpoint={this.getEndpoint()}
+            apiEndpoint={endpoint}
             apiMethod="POST"
             footerClass="modal-footer"
-            onSubmitSuccess={this.onSuccess}
-            submitLabel={t('Reprocess Issue')}
+            onSubmitSuccess={this.handleSuccess}
+            submitDisabled={isFormInvalid}
+            submitLabel={title}
             submitLoadingMessage={t('Reprocessing\u2026')}
             submitErrorMessage={t('Failed to reprocess. Please check your input.')}
-            hideErrors
             onCancel={closeModal}
+            initialData={{maxEvents: ''}}
+            requireChanges
           >
             <NumberField
               name="maxEvents"
-              label={t('Select number of events to be processed')}
+              label={t('Enter the number of events to be reprocessed')}
               help={t(
-                'The default is to reprocess all events, but you can limit the number of events reprocessed in this Issue. If you set a limit, we will reprocess your most recent events.'
+                'You can limit the number of events reprocessed in this Issue. If you set a limit, we will reprocess your most recent events.'
               )}
-              placeholder={t('All events')}
+              placeholder={t('Reprocess all events')}
+              onChange={value => this.handleChangeMaxEvents(value as number)}
```

Comment:
Is the value actually a number here? You could use `parseInt()` to skip the cast and ensure you have the correct type.